File: codewave_py/text_parser.py
import codewave_core.util as util
import codewave_core.editor as editor

class TextParser(editor.Editor):
	def __init__(self,text):
		self.namespace = 'text_parser'
		self._text = text
		self.target = None
	@property
	def text(self):
		return self._text
	@text.setter
	def text(self, val):
		self._text = val
	def textCharAt(self,pos):
		return self.text[pos]
	def textLen(self):
		return len(self.text)
	def textSubstr(self,start, end):
		return self.text[start:end]
	def insertTextAt(self,text, pos):
		self.text = self.text[:pos]+text+self.text[pos:]
	def spliceText(self,start, end, text):
		self.text = self.text[:start]+(text or "")+self.text[end:]
	def getCursorPos(self):
		return self.target
	def setCursorPos(self,start, end = None):
		if end is None:
			end = start
		self.target = util.Pos(start, end)

File: lib/TextParser.js

const Editor = require('./Editor').Editor

const Pos = require('./positioning/Pos').Pos

var TextParser = class TextParser extends Editor {
  constructor (_text) {
    super()
    this._text = _text
  }

  text (val) {
    if (val != null) {
      this._text = val
    }

    return this._text
  }

  textCharAt (pos) {
    return this.text()[pos]
  }

  textLen (pos) {
    return this.text().length
  }

  textSubstr (start, end) {
    return this.text().substring(start, end)
  }

  insertTextAt (text, pos) {
    return this.text(this.text().substring(0, pos) + text + this.text().substring(pos, this.text().length))
  }

  spliceText (start, end, text) {
    return this.text(this.text().slice(0, start) + (text || '') + this.text().slice(end))
  }

  getCursorPos () {
    return this.target
  }

  setCursorPos (start, end) {
    if (arguments.length < 2) {
      end = start
    }

    this.target = new Pos(start, end)
    return this.target
  }
}
exports.TextParser = TextParser
